straxen/plugins/peaklet_processing.py

```diff
@@ -207,6 +225,141 @@ def clip_peaklet_times(peaklets, start, end):
                 p['length'] = (end - p['time']) // p['dt']
 
 
+@numba.jit(nopython=True, nogil=True, cache=True)
+def peak_saturation_correction(records, peaks, to_pe,
+                               reference_length=100,
+                               min_reference_length=20,
+                               use_classification=False,
+                               ):
+    """Correct the area and per pmt area of peaks from saturation
+    :param records: Records
+    :param peaks: Peaklets / Peaks
+    :param reference_length: Maximum number of reference sample used
+    to correct saturated samples
+    :param min_reference_length: Minimum number of reference sample used
+    to correct saturated samples
+    :param use_classification: Choice to use peak type
+    """
+
+    if not len(records):
+        return
+    if not len(peaks):
+        return
+
+    # Search for peaks with saturated channels
+    mask = peaks['n_saturated_channels'] > 0
+    if use_classification:
+        mask &= peaks['type'] == 2
+    peak_list = np.where(mask)[0]
+    # Look up records that touch each peak
+    record_ranges = _touching_windows(
+        records['time'],
+        strax.endtime(records),
+        peaks[peak_list]['time'],
+        strax.endtime(peaks[peak_list]))
+
+    # Create temporary arrays for calculation
+    dt = records[0]['dt']
+    n_channels = len(peaks[0]['saturated_channel'])
+    len_buffer = np.max(peaks['length'] * peaks['dt']) // dt
```

Comment:
The buffer length should be one sample longer is not it? E.g. `length = 1` and `dt_peak = 0.2` would lead to `1*0.2//1 = 0`

Reply:
Right, it is safer to +1. I am under the assumption that dt_peak is multiple of dt_records.